Parser errors in imported modules are silently ignored, causing misleading undefined function errors

## Summary
When an imported module has parser errors, those errors are silently ignored, causing confusing "undefined function" errors instead of showing the actual parser errors.

## Steps to Reproduce
1. Create a module with a parser error (e.g., typo `case` instead of `cast`):

```
// src/codec/serializer.ez
module codec

do serialize_u64(v u64) -> [u8] {
    temp data [byte]
    return case(data, [u8])  // typo: case instead of cast
}
```

2. Import and use the function in main.ez:

```
module main

import & use @std
import codec"./src/codec"

do main() {
    temp result = codec.serialize_u64(257)
    println(result)
}
```

3. Run `ez main.ez`

## Expected Behavior
Should show the parser error from the imported module:
```
error[E2001]: unexpected token '['
  --> src/codec/serializer.ez:5:23
```

## Actual Behavior
Shows a misleading "undefined function" error:
```
error[E4002]: undefined function 'codec.serialize_u64'
  --> main.ez:7:25
```

## Root Cause
In `cmd/ez/main.go` around line 880-882 in the `runFile` function:

```go
if len(fileLex.Errors()) > 0 || fileParser.EZErrors().HasErrors() {
    continue
}
```

Parser errors in imported modules are silently skipped with `continue`, so the errors are never displayed. The module's functions aren't registered, leading to confusing "undefined function" errors.

## Suggested Fix
Instead of silently continuing, print the errors and exit:

```go
if len(fileLex.Errors()) > 0 {
    // Format and print lexer errors
    fmt.Print(errors.FormatErrorList(...))
    os.Exit(1)
}
if fileParser.EZErrors().HasErrors() {
    fmt.Print(errors.FormatErrorList(fileParser.EZErrors()))
    os.Exit(1)
}
```

This would show the actual parser error, helping users identify the real issue.

--- a/cmd/ez/main.go
+++ b/cmd/ez/main.go
@@ -843,7 +843,13 @@ func runFile(filename string) {
 		loader.SetCurrentFile(absPath)
 		mod, err := loader.Load(modPath)
 		if err != nil {
-			continue // Module load errors will be caught at runtime
+			// Check if this is a module error with parse errors (#726)
+			if modErr, ok := err.(*interpreter.ModuleError); ok && modErr.EZErrors != nil && modErr.EZErrors.HasErrors() {
+				fmt.Print(errors.FormatErrorList(modErr.EZErrors))
+				os.Exit(1)
+			}
+			// For other errors (module not found, etc.), let them be caught at runtime
+			continue
 		}
 
 		// For multi-file modules, use two-pass type checking (#722)
@@ -877,8 +883,29 @@ func runFile(filename string) {
 			fileParser := parser.NewWithSource(fileLex, fileSource, filePath)
 			fileProgram := fileParser.ParseProgram()
 
-			if len(fileLex.Errors()) > 0 || fileParser.EZErrors().HasErrors() {
-				continue
+			// Check for lexer errors in imported module
+			if len(fileLex.Errors()) > 0 {
+				errList := errors.NewErrorList()
+				for _, lexErr := range fileLex.Errors() {
+					var code errors.ErrorCode
+					switch lexErr.Code {
+					case "E1005":
+						code = errors.E1005
+					default:
+						code = errors.ErrorCode{Code: lexErr.Code, Name: "lexer-error", Description: "Lexer error"}
+					}
+					sourceLine := errors.GetSourceLine(fileSource, lexErr.Line)
+					ezErr := errors.NewErrorWithSource(code, lexErr.Message, filePath, lexErr.Line, lexErr.Column, sourceLine)
+					errList.AddError(ezErr)
+				}
+				fmt.Print(errors.FormatErrorList(errList))
+				os.Exit(1)
+			}
+
+			// Check for parser errors in imported module
+			if fileParser.EZErrors().HasErrors() {
+				fmt.Print(errors.FormatErrorList(fileParser.EZErrors()))
+				os.Exit(1)
 			}
 
 			parsedFiles = append(parsedFiles, parsedFile{filePath, fileSource, fileProgram})

